### PR TITLE
Implementation Direct OpenMP Implementation Comparison in TPetra SpMV Benchmark

### PR DESCRIPTION
**Pull request by nmhamster**
Adds three tests to TPetra SpMV Benchmark so that comparisons to OpenMP can be made directly:

- **OpenMP Static** which executes SpMV using a static OpenMP schedule
- **OpenMP Dynamic** which executes SpMV using a purely dynamic OpenMP schedule
- **OpenMP Smart-Static** which executes SpMV using a Non-Zero Balancer over threads and a static schedule.

These results were shown way back in July/August but we in custom implementations, this pull-request merges these into the TPetra benchmark so that comparisons between implementations can be made with new compilers and software stacks.

### CODE REVIEW

**Comment by mhoemmen** [comment]
@jjellio and @tjfulle will be interested in this.  I'll take a look after our meeting (in which @nmhamster is also sitting ;-) ).

File: packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp [hunk 1]
```diff
@@ -61,8 +63,11 @@
 #include <Kokkos_SPMV_Inspector.hpp>
 #include <CuSparse_SPMV.hpp>
 #include <MKL_SPMV.hpp>
+#include <OpenMPStatic_SPMV.hpp>
```
Comment (mhoemmen) [inline packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp]: These includes currently won't build if OpenMP is disabled.  Please protect the includes with `#ifdef KOKKOS_HAVE_OPENMP ... #endif`.

File: packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp [hunk 2]
```diff
@@ -103,30 +108,58 @@ int SparseMatrix_generate(OrdinalType nrows, OrdinalType ncols, OrdinalType &nnz
...
-    if(schedule == AUTO)
-      schedule = A.nnz()>10000000?DYNAMIC:STATIC;
-    if(schedule == STATIC)
-      kk_matvec<AType,XType,YType,Kokkos::Static>(A, x, y, rows_per_thread, team_size, vector_length);
-    if(schedule == DYNAMIC)
-      kk_matvec<AType,XType,YType,Kokkos::Dynamic>(A, x, y, rows_per_thread, team_size, vector_length);
-  }
-  if(test == KK_INSP)
-    kk_inspector_matvec(A, x, y, rows_per_thread, team_size, vector_length);
-  #ifdef HAVE_MKL
```
Comment (mhoemmen) [inline packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp]: Please use HAVE_TPETRAKERNELS_MKL.

File: packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp [hunk 3]
```diff
@@ -103,30 +108,58 @@ int SparseMatrix_generate(OrdinalType nrows, OrdinalType ncols, OrdinalType &nnz
...
-  if(test == MKL)
-    mkl_matvec(A, x, y, rows_per_thread, team_size, vector_length);
-  #endif
-  #ifdef HAVE_CUSPARSE
```
Comment (mhoemmen) [inline packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp]: Please use HAVE_TPETRAKERNELS_CUSPARSE.

File: packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp [hunk 4]
```diff
@@ -103,30 +108,58 @@ int SparseMatrix_generate(OrdinalType nrows, OrdinalType ncols, OrdinalType &nnz
...
-  if(test == CUSPARSE)
-    cusparse_matvec(A, x, y, rows_per_thread, team_size, vector_length);
-  #endif
-  #ifdef HAVE_TRILINOS
```
Comment (mhoemmen) [inline packages/tpetra/kernels/perf_test/spmv/test_crsmatrix.cpp]: You're in KokkosKernels.  You ALWAYS have Trilinos.

**Comment by mhoemmen** [comment]
@nmhamster Well, I guess it's merged, so no point in a review ;-P

**Comment by nmhamster** [comment]
@mhoemmen I thought we were just building a small benchmark. Do you want this to be hardened?

**Comment by mhoemmen** [comment]
@nmhamster Sorry I was a bit snarky there :-)  See my last comment on #948.  I would be happy to fix it so that it builds with the CMake build system, but that would involve some small changes to macros etc.  Would y'all object if I did that?

**Comment by nmhamster** [comment]
@mhoemmen no problem. I will fix the code and raise a new PR for you.

**Pull request merged**